NGitLab/Models/GroupId.cs

```diff
@@ -13,24 +24,89 @@ namespace NGitLab.Models;
 
     public GroupId(long id)
     {
-        _id = id;
+        _id = id != 0 ? id : throw new ArgumentException("Id cannot be zero", nameof(id));
     }
 
     public GroupId(string path)
     {
         _path = path ?? throw new ArgumentNullException(nameof(path));
     }
 
-    public GroupId(Group group)
```

Comment:
Removed this c'tor because the resulting `GroupId` is ambiguous. For example,
```C#
Group group = GetGroup(...);
var id1 = new GroupId(group); // calls the removed c`tor
var id2 = new GroupId(group.Id);
var id3 = new GroupId(group.FullPath);

Assert(id1 == id2);  // Ok, both contain `group.Id`
Assert(id1 == id3);  // Fails! group.Id != group.FullPath.
Assert(id2 == id3);  // Fails for same reason, but there no easy way around this other than to not do it.

// Preferred usage:
Assert(id1 == group); // Ok!
Assert(id2 == group); // Ok!
Assert(id3 == group); // Ok!
```

Reply:
It's probably wise to eventually get rid of this ctor, but for now please leave it in and tag is with the `[Obsolete]` attribute, so as not to introduce a breaking change.